buildSrc/src/main/kotlin/ktlint-kotlin-common.gradle.kts

```diff
@@ -22,7 +22,7 @@ kotlin {
 tasks.withType<KotlinCompile>().configureEach {
     kotlinOptions {
         @Suppress("SuspiciousCollectionReassignment")
-        freeCompilerArgs += listOf("-opt-in=kotlin.RequiresOptIn")
+        freeCompilerArgs += listOf("-Xuse-k2")
```

Comment:
https://kotlinlang.org/docs/whatsnew17.html#new-kotlin-k2-compiler-for-the-jvm-in-alpha

Reply:
I am sure whether we should use the Alpha K2 compiler already, but I am willing to give it a go.